desktop-src/direct3dhlsl/WaveMatch.md

```diff
@@ -0,0 +1,76 @@
+nms.assetid:
+title: WaveMatch
+ms.topic: reference
+ms.date: 07/10/2024
+topic_type:
+- APIRef
+- kbSyntax
+api_name:
+- WaveMatch
+api_type:
+- NA
+---
+
+
+# WaveMatch
+
+Checks if all lanes in the wave have the same value.
+
+## Syntax
+
+
+```syntax
+uint<4> WaveMatch(numeric<> value);
+```
+
+## Parameters
+
+| Item | Description |
+|------|-------------|
+| *value* | [in] An input parameter of vector type. This input value is used as the basis for the WaveMatch operation.  |
+## Return value
+
+ The function returns a vector of type uint. Each component of the returned vector represents a binary mask indicating which lanes of the wave have the same value as the called lane.
+## Type Description
+
+| Name  | [**Template Type**](../direct3dhlsl/dx-graphics-hlsl-data-types.md)| [**Component Type**](../direct3dhlsl/dx-graphics-hlsl-data-types.md) | Size |
+|-------|--------------------------------------------------------------------|----------------------------------------------------------------------|------|
+| *ret*   | [**vector**](../direct3dhlsl/dx-graphics-hlsl-vector.md) | [**uint**](../WinProg/windows-data-types) | 4 |
+| *value*   | [**scalar**](../direct3dhlsl/dx-graphics-hlsl-scalar.md), [**vector**](../direct3dhlsl/dx-graphics-hlsl-vector.md), or [**matrix**](../direct3dhlsl/dx-graphics-hlsl-matrix.md) | [**float**](../WinProg/windows-data-types) or [**int**](../WinProg/windows-data-types) | any |
+
+## Minimum Shader Model
+
+This function is supported in the following shader models.
+|Shader Model |	Supported|
+|-------------|----------|
+|[Shader Model 6.5](https://microsoft.github.io/DirectX-Specs/d3d/HLSL_ShaderModel6_5) and higher shader models | yes |
+
+## Shader Stages
+
+* **Library Shader**
+* [**Compute Shader**](../direct3d11/direct3d-11-advanced-stages-compute-shader.md)
+* [**Amplification Shader**](https://microsoft.github.io/DirectX-Specs/d3d/MeshShader.html#amplification-shader-and-mesh-shader)
+* [**Mesh Shader**](https://microsoft.github.io/DirectX-Specs/d3d/MeshShader.html)
+* [**Pixel Shader**](../direct3dhlsl/dx-graphics-hlsl-writing-shaders-9.md#pixel-shader-basics)
+* [**Vertex Shader**](../direct3dhlsl/dx-graphics-hlsl-writing-shaders-9.md#vertex-shader-basics)
+* [**Hull Shader**](https://learn.microsoft.com/en-us/windows/uwp/graphics-concepts/hull-shader-stage--hs-)
+* [**Domain Shader**](https://learn.microsoft.com/en-us/windows/uwp/graphics-concepts/domain-shader-stage--ds-)
+* [**Geometry Shader**](https://learn.microsoft.com/en-us/windows/uwp/graphics-concepts/geometry-shader-stage--gs-)
+* [**Raygeneration Shader**](../direct3d12/ray-generation-shader.md)
+* [**Intersection Shader**](../direct3d12/intersection-shader.md)
+* [**Anyhit Shader**](../direct3d12/any-hit-shader.md)
+* [**Closesthit Shader**](../direct3d12/closest-hit-shader.md)
+* [**Miss Shader**](../direct3d12/miss-shader.md)
+* [**Callable Shader**](../direct3d12/callable-shader.md)
+* **Node Shader**
```

Comment:
This isn't an llm failure. Some intrinsics have documentation about which shader models are supported and which aren't that is because DXILOps 6-71, 78-80, 101, 130-131, 160, 162-164,  216-221, and 244 never had shader model documentation while the rest do.